pkg/controllers/job/job_controller.go

```diff
@@ -215,20 +239,66 @@ func (cc *Controller) Run(stopCh <-chan struct{}) {
 	glog.Infof("JobController is running ...... ")
 }
 
-func (cc *Controller) worker() {
-	for cc.processNextReq() {
+func (cc *Controller) worker(i uint32) {
+	glog.Infof("worker %d start ...... ", i)
+
+	for cc.processNextReq(i) {
 	}
 }
 
-func (cc *Controller) processNextReq() bool {
-	obj, shutdown := cc.queue.Get()
+func (cc *Controller) belongsToThisRoutine(key string, count uint32) bool {
+	var hashVal hash.Hash32
+	var val uint32
+
+	hashVal = fnv.New32()
+	hashVal.Write([]byte(key))
+
+	val = hashVal.Sum32()
+
+	if val%cc.workers == count {
+		return true
+	}
+
+	return false
+}
+
+func (cc *Controller) getWorkerQueue(key string) workqueue.RateLimitingInterface {
+	var hashVal hash.Hash32
+	var val uint32
+
+	hashVal = fnv.New32()
+	hashVal.Write([]byte(key))
+
+	val = hashVal.Sum32()
+
+	cc.lock.Lock()
+	queue := cc.queueList[val%cc.workers]
+	cc.lock.Unlock()
+
+	return queue
+}
+
+func (cc *Controller) processNextReq(count uint32) bool {
+	cc.lock.Lock()
+	queue := cc.queueList[count]
```

Comment:
Is the lock necessary?

Reply:
yeah, I felt as the queueList is accessed by multiple threads paralelly

Reply:
nope ， concurrent access an array does not cause race.

Reply:
there will be no write ops after the start, right?

Reply:
yeah, no write ops, may be i miss understood ur previous comment

hzxuzhonghu 23 days ago  Member
nit: I would like to acquire the queue before this loop

Reply:
hha

Reply:
>nit: I would like to acquire the queue before this loop

Just ignore this now

Reply:
changed